File: Week2/BlueStaxks/2-3.cpp
#include <string>
#include <vector>
using namespace std;
vector<int> solution(vector<int> progresses, vector<int> speeds)
{
    vector<int> answer;
    int s = (int)speeds.size(), i, p = 0, c;
    while (1)
    {
        c = 0; //배포할 기술 개수
        if (progresses[p] >= 100)
        {
            while (progresses[p] >= 100)
            {
                c++;
                p++; //순서대로 배포해야 하기 때문에 증가만 하는 p 사용
                if (p >= s) //끝까지 다 만들면 끝
                {
                    answer.push_back(c);
                    return answer;
                }
            }
            answer.push_back(c);
        }
        for (i = p; i < s; ++i)
            progresses[i] += speeds[i]; //개발 진행
    }
}

File: Week2/heystranger/2-3.py
def solution(progresses, speeds):
    day = []  # 작업 처리소요 기간
    out = []  # 배포되는 날
    answer = []

    for i in range(len(progresses)):
        if (100 - progresses[i]) % speeds[i] != 0:
            day.append((100 - int(progresses[i])) / int(speeds[i]) + 1)
        else:
            day.append((100 - int(progresses[i])) / int(speeds[i]))

    for k in range(len(progresses)):
        if k == 0:
            out.append(int(day[i]))
        else:
            if int(out[k - 1]) < int(day[k]):
                out.append(int(out[k]) + 1)
            else:
                out.append(int(day[k]))

    for j in range(len(out)):
        if j == 0:
            answer.append(1)

        else:
            if answer[j - 1] == out[j]:
                answer[j] = (int(answer[j - 1]) + 1)
            else:
                answer.append(1)

    return answer